lib/Mojo/Promise.pm

```diff
@@ -41,6 +41,31 @@ sub finally {
   return $new;
 }
 
+sub map {
+  my ($class, $options) = (shift, ref $_[0] eq 'HASH' ? shift : {});
+  my ($cb, @items) = @_;
+
+  my @start = map { $class->resolve($_->$cb) } splice @items, 0,
```

Comment:
I don't think this will work, since resolve() on the class use the singleton ioloop and not the IOLoop from the item(s) you pass in.

Reply:
Maybe it's better to require that the map-function only return promises, since that's the requirement for the input to all()?

Reply:
> Maybe it's better to require that the map-function only return promises, since that's the requirement for the input to all()?

That was my original intent.